call-notes/Epoch Rewards.md

```diff
@@ -0,0 +1,171 @@
+
+**Attendee 1**:  So basically what you have been watching has been reached when all the distribution is completed, and then make that call. Is that what you're suggesting? 
+
+
+**Attendee 3**: It’s deterministic, Yeah, you know, it's over by the thousandth block after the epoch, so you can just make the call right there instead, right?
+
+**Attendee 2**:  So today, where you just know that it's, you know, the first block in the epoch, now it's going to be, you know, the thousandth block, or that, you know, however many blocks before. I don't think any of this is fully parameterized yet. 
+
+**Attendee 1**: Okay, and so, well, so someone would have to have sort of code that understands that to be able to compute what that block would be, or there'll be an RPC call to ask that at some point. Like, is that how you anticipate it going? 
+
+
+**Attendee 3**: Yeah, basically. It'll depend on how much complexity there ends up being in the final thing. I don't anticipate it being that difficult. This could probably be like an RPC helper versus an actual RPC endpoint, but I think it'll be fairly trivial to put together a way to equivalently get, I mean, admittedly, it's going to be a little heavier because you have to get multiple blocks, right? 
+
+
+**Attendee 1** :  Okay. And I do have another question, but I don't want to monopolize this, so I'll wait to see if other people have a question first.
+
+**Jacob Creech**: All right, next question.
+
+I think it was Jeff, so go ahead, Jeffy, you should be able to unmute to ask your question. 
+
+**Jeff** : Oh, sorry, my mistake. 
+
+**Jacob Creech**: Okay, is there any other questions about this, or otherwise we'll go back to you, Zans. All right, Ashwin, go ahead.
+
+**Aswin**: Hey just wanted to ask, if there's a cluster we started in the first thousand blocks, is there any change to change that procedure because the snapshot won't have the result in it? 
+
+**Haoran** : Yeah, we have a talk about this, so in that case, when the snapshot is taken during the reward computation phase, the snapshot has to wait, it has to wait until the result is available and store the result there.
+
+**Aswin**: Got it. 
+
+**Jacob Creech**: Okay, Zans, you're welcome to ask your question. 
+
+**Zantetsu**: Okay, so is it the case that validators will only validate a block if the leader producing that block has the appropriate amount of either computation work or rewards pair of work in that block? Or is it the case that a validator can just sort of decide not to do it on their block and be like, "I've modified the code to not do this because I don't want to burn the CPU on this and it doesn't give me any rewards, so I'm just not going to do it?" Or is that even a possibility? Or is it the case that the cluster won't even accept a block unless it includes these details because they're expected for... I mean, they're sort of built into the protocol that this has to happen in this order. 

```

Comment:
```suggestion
**Zantetsu**: Okay, so is it the case that validators will only validate a block if the leader producing that block has the appropriate amount of either computation work or rewards payout of work in that block? Or is it the case that a validator can just sort of decide not to do it on their block and be like, "I've modified the code to not do this because I don't want to burn the CPU on this and it doesn't give me any rewards, so I'm just not going to do it?" Or is that even a possibility? Or is it the case that the cluster won't even accept a block unless it includes these details because they're expected for... I mean, they're sort of built into the protocol that this has to happen in this order. 

```